libraries/Adapters/Microsoft.Bot.Builder.Adapters.Twilio/TwilioHelper.cs

```diff
@@ -0,0 +1,142 @@
+        {
+            var attachments = new List<Attachment>();
+            if (int.TryParse(message.NumMedia, out var numMediaResult) && numMediaResult > 0)
+            {
+                for (var i = 0; i < numMediaResult; i++)
+                {
+                    var attachment = new Attachment()
+                    {
+                        ContentType = message.MediaContentTypes[i],
+                        ContentUrl = message.MediaUrls[i].AbsolutePath,
+                    };
+                    attachments.Add(attachment);
+                }
+            }
+
+            return attachments;
+        }
+
+        /// <summary>
+        /// Converts a query string to a dictionary with key-value pairs.
+        /// </summary>
+        /// <param name="query">The query string to convert.</param>
+        /// <returns>A dictionary with the query values.</returns>
+        public static Dictionary<string, string> QueryStringToDictionary(string query)
+        {
+            var pairs = query.Replace("+", "%20").Split('&');
+            var values = new Dictionary<string, string>();
+
+            foreach (var p in pairs)
+            {
+                var pair = p.Split('=');
+                var key = pair[0];
+                var value = Uri.UnescapeDataString(pair[1]);
+
+                values.Add(key, value);
+            }
+
+            return values;
+        }
+
+        /// <summary>
+        /// Formats a BotBuilder activity into an outgoing Twilio SMS message.
+        /// </summary>
+        /// <param name="activity">A BotBuilder Activity object.</param>
+        /// <param name="options">A set of params with the required values for authentication.</param>
+        /// <returns>A Message's options object with {body, from, to, mediaUrl}.</returns>
+        public static CreateMessageOptions ActivityToTwilio(Activity activity, ITwilioAdapterOptions options)
+        {
+            var mediaUrls = new List<Uri>();
+
+            if ((activity.ChannelData as TwilioEvent)?.MediaUrls != null)
+            {
+                mediaUrls = ((TwilioEvent)activity.ChannelData).MediaUrls;
+            }
+
+            var messageOptions = new CreateMessageOptions(activity.Conversation.Id)
+            {
+                ApplicationSid = activity.Conversation.Id,
+                From = options.TwilioNumber,
+                Body = activity.Text,
+                MediaUrl = mediaUrls,
+            };
+
+            return messageOptions;
+        }
+
+        /// <summary>
+        /// Processes a HTTP request into an Activity.
+        /// </summary>
+        /// <param name="httpRequest">A httpRequest object.</param>
+        /// <param name="options">A set of params with the required values for authentication.</param>
+        /// <returns>The Activity obtained from the httpRequest object.</returns>
+        public static Activity ReadRequest(HttpRequest httpRequest, ITwilioAdapterOptions options)
+        {
```

Comment:
Shouldn't we check if _httpRequest_ and _options_ are _**null**_?
I think that in case any of these is null, we should return an _empty Activity_ (or should we return _null?)._ What do you think about this?